### ISSUE TITLE
s3_upload_path errors

### ISSUE THREAD

**Issue by Keno** [JuliaCloud/AWSS3.jl#84]
```
type Array has no field headers
Stacktrace:
 [1] getproperty(::Array{UInt8,1}, ::Symbol) at ./Base.jl:33
 [2] s3_upload_part(::Dict{Symbol,Any}, ::XMLDict.XMLDictElement, ::Int64, ::Array{UInt8,1}) at /home/keno/.julia/dev/AWSS3/src/AWSS3.jl:697
```

Looks like this function expects to get a raw response object back from the API, but instead only gets the body.

**Comment by mattBrzezinski** [comment]
Looking into this I can see where this issue is stemming from.

```julia
function AWSS3.s3_upload_part(aws::AWSConfig, upload, part_number, part_data)
    response = s3(aws, "PUT", upload["Bucket"];
                  path = upload["Key"],
                  query = Dict("partNumber" => part_number,
                               "uploadId" => upload["UploadId"]),
                  content = part_data)

    response.headers["ETag"]  # Code breaks here
end
```

```julia
function AWSS3.s3()
    ....
    response = AWSCore.do_request(request)
    ....
```

```julia
function AWSCore.do_request(r::AWSRequest)
    ....
    return response.body
    ....
```

The only time headers get returned from `AWSCore` is if a `HEAD` request is performed, [code](https://github.com/JuliaCloud/AWSCore.jl/blob/master/src/AWSCore.jl#L498).

**Comment by mattBrzezinski** [comment]
@Keno do you require the `ETag`, or any of the other [common response headers](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html)? If not I propose that we modify `s3_upload_part()` just return the `response` for now.

Alternatively, we can release a new version of `AWSCore` which `do_request()` returns back the full `response` object by default, and version bump `AWSS3` to match.

**Comment by mattBrzezinski** [comment]
Resolved in #86
